### ISSUE TITLE
[REGRESSION] argument(s) mismatch for an intrinsic 'trim()'.

### ISSUE THREAD

**Issue by clementval** [omni-compiler/omni-compiler#492]
This code used to work but it is no more the case.

```fortran
MODULE mod_issue492
  CHARACTER ( LEN= 80 ) , PUBLIC , EXTERNAL :: nf_strerror
END MODULE mod_issue492
```

```fortran
MODULE mod_issue492_2
  USE mod_issue492, only: nf_strerror

CONTAINS
  SUBROUTINE sub1()
    print*, trim(nf_strerror(0))
  END SUBROUTINE sub1
END MODULE mod_issue492_2
```

PR #490 should have caused the regression

**Comment by clementval** [comment]
@shingo-s your PR #490 made this code not passing anymore. Any idea why before I start debugging it?

**Comment by shingo-s** [comment]
@clementval 

In my environment (gfortran-6 and fc287a733c66c66166bc5e185c108a2c609e768c), the above code doesn't show any error (excepts `nf_strerror_` is undefined).

Do you have any error messages?

And PR #490 contains F-BackEnd update, so please make sure the jar file is updated.

**Comment by clementval** [comment]
@shingo-s I git this error `"dummy2.f90:mod_issue492_2", line 6: argument(s) mismatch for an intrinsic 'trim()'.`

I put the first piece of code in a file and the second in another file. 
I just check out the latest master and still have the error.

**Comment by clementval** [comment]
@shingo-s If I go back to 1e50e67ba3ea1189d57c47fb8e178e9ea463867c the code is fine. 
The major difference I have seen so far is in the `.xmod` file. the attribute `is_public="true"` has disappeared from the `FfunctionType`.

**Comment by clementval** [comment]
@shingo-s I could narrow down the problem to this commit 816dbef43c59afcb7980ca6d89006aa10d73fcab

**Comment by clementval** [comment]
So the problem seems to be a new indirection in the type definition. 
This `.xmod` before 816dbef43c59afcb7980ca6d89006aa10d73fcab is working properly.
```xml
<OmniFortranModule version="1.0">
  <name>mod_issue492</name>
  <depends>
  </depends>
  <typeTable>
    <FfunctionType type="F7fb764c08260" return_type="C7fb764c08140" is_external="true" is_public="true"/>
    <FbasicType type="C7fb764c07f40" ref="Fcharacter">
      <len>
        <FintConstant type="Fint">80</FintConstant>
      </len>
    </FbasicType>
    <FbasicType type="C7fb764c08140" ref="C7fb764c07f40"/>
  </typeTable>
  <identifiers>
    <id type="F7fb764c08260" sclass="ffunc" original_name="nf_strerror" declared_in="mod_issue492">
      <name>nf_strerror</name>
    </id>
  </identifiers>
  <declarations>
  </declarations>
  <interfaceDecls>
  </interfaceDecls>
  <aux_info>
  </aux_info>
</OmniFortranModule>
```

This is not working anymore. 
```xml
<OmniFortranModule version="1.0">
  <name>mod_issue492</name>
  <depends>
  </depends>
  <typeTable>
    <FfunctionType type="F7fd1eb4086f0" return_type="C7fd1eb408380" is_external="true"/>
    <FbasicType type="C7fd1eb4084a0" ref="Fcharacter">
      <len>
        <FintConstant type="Fint">80</FintConstant>
      </len>
    </FbasicType>
    <FbasicType type="C7fd1eb408380" ref="C7fd1eb4084a0"/>
    <FbasicType type="F7fd1eb408810" is_public="true" ref="F7fd1eb4086f0"/>
  </typeTable>
  <identifiers>
    <id type="F7fd1eb408810" sclass="ffunc" original_name="nf_strerror" declared_in="mod_issue492">
      <name>nf_strerror</name>
    </id>
  </identifiers>
  <declarations>
  </declarations>
  <interfaceDecls>
  </interfaceDecls>
  <aux_info>
  </aux_info>
</OmniFortranModule>
```

**Comment by shingo-s** [comment]
@clementval 

New .xmod is vailid and `import_module_id()` seems to work fine, so there is a bug in `compile_function_call()`.

Please fix F-compile-expr.c:2433
```c
            if (!IS_PROCEDURE_TYPE(tp) ||
                (IS_PROCEDURE_POINTER(tp) && !TYPE_IS_EXTERNAL(tp))) {
                tp = function_type(tp);
                ID_TYPE(f_id) = tp;
                EXPV_TYPE(ID_ADDR(f_id)) = ID_TYPE(f_id);
            }
```

For the sample, `tp` is the type of `nf_strerror`. And the above code expects `tp` has a `TYPE_EXTERANL_ATTR`, but actually `TYPE_EXTERNAL_ATTR` is owned by `TYPE_REF(tp)`.

**Comment by clementval** [comment]
@shingo-s Thanks for the hint! Should be fixed with PR #512
